src/main/java/com/google/cloud/tools/gradle/appengine/core/CheckCloudSdkTask.java

```diff
@@ -44,12 +44,11 @@ public void setCloudSdkBuilderFactory(CloudSdkBuilderFactory cloudSdkBuilderFact
   /** Task entrypoint : Verify Cloud SDK installation. */
   @TaskAction
   public void checkCloudSdkAction() {
-    File home = toolsExtension.getCloudSdkHome();
-    if (home == null) {
-      throw new GradleException("SDK home directory must be specified for validation.");
+    if (Strings.isNullOrEmpty(version) || cloudSdkBuilderFactory == null) {
```

Comment:
If we make `version` an `@Input` https://docs.gradle.org/current/javadoc/org/gradle/api/tasks/Input.html and the gradle task will fail automatically when this isn't configured.

Reply:
`@Input` is supposed to be used on getters, but these parameters don't have getters (only setters, which are ignored by `@Input`). I don't think it's worth it to add getters just for this.

Reply:
Yeah, I mean if we wrote the plugin in groovy you would throw it on the property directly. But for this case, it's fine, using a getter is the standard way to define task inputs.